src/shared/containers/tc-communities/Header.jsx

```diff
@@ -15,13 +15,23 @@ function mapStateToProps(state, ownProps) {
    * meta data are loaded into Redux store. Thus, no need to make any checks
    * of "meta" object here, we can rely it exists and is properly loaded. */
   const meta = state.tcCommunities.meta.data;
+  const selectorLabels = meta.communitySelector.map(({ label }) => label);
+  const communitySelector = meta.communitySelector.concat(
+    state.tcCommunities.list.filter(({ communityId, communityName }) =>
+      communityId !== meta.communityId && selectorLabels.indexOf(communityName) < 0)
+      .map(({ communityId, communityName }, i) => ({
+        value: (i + (meta.communitySelector || []).length).toString(),
+        label: communityName,
+        redirect: `/community/${communityId}`,
+      })),
+  );
```

Comment:
This block of code looks correct, most probably, but it is placed in the wrong place: never ever create new objects inside `mapStateToProps()` to further pass them inside the component!!! If you do this, each time React re-evaluates the state, it will call this `mapStateToProps()`; it will create a new object for `communitySelector` (even when no state changes happened at all) and pass it inside React container; react will see that the object has been changed (it does shallow comparison) and will unnecessarily re-render entire sub-tree of components.

The correct way: pass entire `state.tcCommunities.list` inside the component, and then do any further filtering / mapping inside the render method, directly responsible for the dropdown. This way we avoid unnecessary re-renderings, and ensure good performance of the app.